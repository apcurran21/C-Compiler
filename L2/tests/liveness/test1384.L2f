(@add
	7
	%sum <- rdi
	%sum += rsi
	%sum += rdx
	%sum += rcx
	%sum += r8
	%sum += r9
	%arg7 <- stack-arg 0
	%sum += %arg7
	%sum <<= 1
	%sum++
	rdi <- %sum
	call print 1
	rax <- %sum
	return
)
