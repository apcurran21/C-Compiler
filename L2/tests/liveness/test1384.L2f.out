(
(in
(r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi)
(r12 r13 r14 r15 r8 r9 rbp rbx rcx rdx rsi %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx rcx rdx %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx rcx %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx %sum)
(r12 r13 r14 r15 r9 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(%arg7 r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx rdi %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(r12 r13 r14 r15 r8 r9 rbp rbx rcx rdx rsi %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx rcx rdx %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx rcx %sum)
(r12 r13 r14 r15 r8 r9 rbp rbx %sum)
(r12 r13 r14 r15 r9 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(%arg7 r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rbp rbx rdi %sum)
(r12 r13 r14 r15 rbp rbx %sum)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
